### ISSUE TITLE
QQuickWidget: Cannot render due to failing makeCurrent()

### ISSUE THREAD

**Issue by bunwayenterprises** [qutebrowser/qutebrowser#5251]
**Version info (see `:version`)**:

v1.10.1

**Does the bug happen if you start with `--temp-basedir`?**:

Yes (as far as I can tell)

**Description**

Currently running on arch with sway 1.4.

Qutebrowser runs for a while (it can be a few hours or a couple of days) and over time when switching back and forth between workspaces, the full qutebrowser window is rendered blank (no tabs, no content, no statusbar or command input elements). It is still responsive as I can use the shortcuts to interact with it (ie - open new tabs) but it just renders blank and needs to be restarted.

The following output is found on the command line when running as a foreground process:

```
15:07:25 WARNING: QPaintDevice::metrics: Device has no metric information
15:07:25 WARNING: QQuickWidget: Cannot render due to failing makeCurrent()
15:07:25 WARNING: QPaintDevice::metrics: Device has no metric information
15:07:25 WARNING: QPaintDevice::metrics: Device has no metric information
15:07:25 WARNING: QQuickWidget: Cannot render due to failing makeCurrent()
15:07:25 WARNING: QQuickWidget: Cannot render due to failing makeCurrent()
15:12:54 WARNING: QPaintDevice::metrics: Device has no metric information
15:12:54 WARNING: QPaintDevice::metrics: Device has no metric information
15:12:54 WARNING: QQuickWidget: Cannot render due to failing makeCurrent()
15:12:54 WARNING: QQuickWidget: Cannot render due to failing makeCurrent()
15:12:54 WARNING: QQuickWidget: Cannot render due to failing makeCurrent()
15:12:54 WARNING: QQuickWidget: Cannot render due to failing makeCurrent()
15:12:57 WARNING: QQuickWidget: Cannot render due to failing makeCurrent()
15:25:03 WARNING: QQuickWidget: Cannot render due to failing makeCurrent()
15:25:04 WARNING: QQuickWidget: Cannot render due to failing makeCurrent()
```

Currently running with the following wrapper script:

```
#!/bin/bash

export QT_AUTO_SCREEN_SCALE_FACTOR=0
export QT_SCALE_FACTOR=1.5
export QT_QPA_PLATFORM=wayland-egl

qutebrowser --target tab-bg-silent $1
```

**Comment by The-Compiler** [comment]
I don't think there's anything qutebrowser can do about this - it looks like a bug in either Qt or your graphics driver.

What Qt version are you on?

You might want to play with the different `qt.force_software_rendering` options and see if that changes anything.

**Comment by bunwayenterprises** [comment]
Currently on QT 5.14.1

Thanks for the suggestion. I'll investigate intel drivers and try with software rendering as well.

**Comment by The-Compiler** [comment]
FWIW this looks similar: [[QTBUG-78461] [regression in 5.14] WebEngine fails to recover after suspend - Qt Bug Tracker](https://bugreports.qt.io/browse/QTBUG-78461)

However, that should've been fixed in Qt 5.14.0.

If you can reproduce this with another QtWebEngine browser such as [Falkon](https://www.falkon.org) as well, you might want to open a [Qt bug](https://bugreports.qt.io/secure/Dashboard.jspa).

Closing this now as I don't think this is something qutebrowser can fix - sorry!
